src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -2178,13 +2189,27 @@ protected boolean compileProgram() {
 
             packageName = "";
             packagePath = "";
-            String programName = replaceString(language.getExecutableIdentifierMask(), "{:basename}", removeExtension(runFiles.getMainFile().getName()));
+            
+            String programName = language.getExecutableIdentifierMask();
+            
+            // the "executable" program name is the entry point, if one exists, so try to substitute that first
+            if (run.getEntryPoint() != null) {
```

Comment:
Same comment as above; I think this should also test for `!isBlank()` and `!isEmpty()`.

Reply:
Fixed.  Used @lane55's recommendation of using StringUtilities.isEmpty().